examples/mnist/pipeline/mnist_spark.py

```diff
@@ -0,0 +1,120 @@
+parser.add_argument("--model_dir", help="HDFS path to save/load model during train/inference", default="mnist_model")
+parser.add_argument("--export_dir", help="HDFS path to export model", default="mnist_export")
+parser.add_argument("--cluster_size", help="number of nodes in the cluster", type=int, default=num_executors)
+parser.add_argument("--num_ps", help="number of PS nodes in cluster", type=int, default=1)
+parser.add_argument("--rdma", help="use rdma connection", action="store_true")
+parser.add_argument("--steps", help="maximum number of steps", type=int, default=1000)
+parser.add_argument("--tensorboard", help="launch tensorboard process", action="store_true")
+
+######## ARGS ########
+
+# Spark input/output
+parser.add_argument("--format", help="example format: (csv|pickle|tfr)", choices=["csv","pickle","tfr"], default="csv")
+parser.add_argument("--images", help="HDFS path to MNIST images in parallelized format")
+parser.add_argument("--labels", help="HDFS path to MNIST labels in parallelized format")
+parser.add_argument("--output", help="HDFS path to save test/inference output", default="predictions")
+
+args = parser.parse_args()
+print("args:",args)
+
+print("{0} ===== Start".format(datetime.now().isoformat()))
+
+if args.format == "tfr":
+  images = sc.newAPIHadoopFile(args.images, "org.tensorflow.hadoop.io.TFRecordFileInputFormat",
+                              keyClass="org.apache.hadoop.io.BytesWritable",
+                              valueClass="org.apache.hadoop.io.NullWritable")
+  def toNumpy(bytestr):
+    example = tf.train.Example()
+    example.ParseFromString(bytestr)
+    features = example.features.feature
+    image = numpy.array(features['image'].int64_list.value)
+    label = numpy.array(features['label'].int64_list.value)
+    return (image, label)
+  dataRDD = images.map(lambda x: toNumpy(str(x[0])))
+else:
+  if args.format == "csv":
+    images = sc.textFile(args.images).map(lambda ln: [int(x) for x in ln.split(',')])
+    labels = sc.textFile(args.labels).map(lambda ln: [float(x) for x in ln.split(',')])
+  else: # args.format == "pickle":
+    images = sc.pickleFile(args.images)
+    labels = sc.pickleFile(args.labels)
+  print("zipping images and labels")
+  dataRDD = images.zip(labels)
+
+# Pipeline API
+df = spark.createDataFrame(dataRDD)
+
+print("{0} ===== Estimator.fit()".format(datetime.now().isoformat()))
+# dummy tf args (from imagenet/inception example)
+tf_args = { 'initial_learning_rate': 0.045, 'num_epochs_per_decay': 2.0, 'learning_rate_decay_factor': 0.94 }
+estimator = TFEstimator(mnist_dist.map_fun, tf_args) \
+        .setModelDir(args.model_dir) \
+        .setExportDir(args.export_dir) \
+        .setClusterSize(args.cluster_size) \
+        .setNumPS(args.num_ps) \
+        .setRDMA(args.rdma) \
+        .setTensorboard(args.tensorboard) \
+        .setEpochs(args.epochs) \
+        .setBatchSize(args.batch_size) \
+        .setSteps(args.steps)
+
+model = estimator.fit(df)
```

Comment:
In general, df will have multiple columns. We need to specify a map from input columns to the corresponding input tensor aliases (ex, 'input', 'label').